tests/test_encode.py

```diff
@@ -508,6 +508,33 @@ def test_encode_type_19():
     assert encoded[0] == "!AIVDO,1,1,,A,C5N3SRP0=nJGEBT>NhWAwwo862PaLELTBJ:V0000000000D:R220,0*46"
 
 
+def test_encode_type_18_with_speed_and_course():
+    data = {
+        'accuracy': 0,
+        'assigned': 0,
+        'band': 1,
+        'course': 10.1,
+        'cs': 1,
+        'display': 0,
+        'dsc': 1,
+        'heading': 511,
+        'lat': 37.785035,
+        'lon': -122.26732,
+        'mmsi': '367430530',
+        'msg22': 1,
+        'radio': 917510,
+        'raim': 0,
+        'regional': 0,
+        'repeat': 0,
+        'second': 55,
+        'speed': 67.85,
```

Comment:
speed should not have 2 decimal places, only one.

Reply:
That's part of the testcase actually. This testcase tests the encoding of a message. And a user might pass a float that has more than a single decimal place. But this should be handled by the lib - which it is.

If you decode `!AIVDO,1,1,,A,B5NJ;PP2aUl4ot5Isbl6GwsUkP06,0*35` you get `67.8`.